File: 1829-maximum-units-on-a-truck/maximum-units-on-a-truck.java
class Solution {
    public int maximumUnits(int[][] boxTypes, int truckSize) {
        int ans = 0;

        Arrays.sort(boxTypes, (a, b) -> Integer.compare(b[1], a[1]));

        for (int[] box : boxTypes) {
            int boxCount = Math.min(truckSize, box[0]);
            ans += boxCount * box[1];
            truckSize -= boxCount;

            if (truckSize == 0)
                break;
        }

        return ans;
    }
}

File: 1829-maximum-units-on-a-truck/maximum-units-on-a-truck.js
/**
 * @param {number[][]} boxTypes
 * @param {number} truckSize
 * @return {number}
 */
var maximumUnits = function (boxTypes, truckSize) {
    boxTypes.sort((a, b) => b[1] - a[1]);

    let maxUnits = 0;

    for (let i = 0; i < boxTypes.length && truckSize > 0; i++) {
        const [boxCount, unitsPerBox] = boxTypes[i];
        const takeBoxes = Math.min(truckSize, boxCount);
        truckSize -= takeBoxes;
        maxUnits += takeBoxes * unitsPerBox;
    }

    return maxUnits;
};
